tests/test.collector.js

```diff
@@ -65,6 +65,30 @@ describe('Collector', function() {
     assert.equal(collection.notices.length, 0);
   });
 
+  it('should not add a duplicate message with a dataPath', () => {
+    var collection = new Collector();
+    collection.addWarning({ ...fakeMessageData, dataPath: '/foo' });
+    collection.addWarning({ ...fakeMessageData, dataPath: '/foo' });
+    assert.equal(collection.warnings.length, 1);
+    assert.equal(collection.warnings[0].type, 'warning');
+    assert.equal(collection.errors.length, 0);
+    assert.equal(collection.notices.length, 0);
+  });
+
+  it('should add a message that differs on one prop', () => {
+    var collection = new Collector();
+    collection.addWarning({ ...fakeMessageData, dataPath: '/foo' });
+    collection.addWarning({
+      ...fakeMessageData,
+      dataPath: '/foo',
+      message: 'Foo message',
+    });
```

Comment:
This explicitly only tests `message` but for regular messages, say, some kind of warning that happens like 10x in a file mostly only the `line` is different afaik (I could be completely wrong though). So can we maybe add a test and make sure that we don't de-duplicate messages where only the `line` (or other properties) are because we should keep them.